tests/api/testdata/getters.py

```diff
@@ -29,11 +29,15 @@ def get_endpoint_data(adcm: ADCMTestApiWrapper, endpoint: Endpoints) -> list:
         request=Request(endpoint=endpoint, method=Methods.LIST),
         expected_response=ExpectedResponse(status_code=Methods.LIST.value.default_success_code),
     )
-    if isinstance(res.json(), list):
-        return res.json()
+    result = res.json()
     # New endpoints always return a response with pagination.
     # In the future all endpoints will return that
-    return res.json().get("results")
+    if not isinstance(result, list):
+        result = result.get("results")
+    if endpoint.endpoint.filter_predicate is None:
+        return result
+    # Is used mostly for "special" or "corner-case" endpoints
```

Comment:
not really. it is used for endpoints with more than one dataclass
"special or corner-case" does not explain

Reply:
Changed